acceptance/tests/resource/service/smf_basic_tests.rb

```diff
@@ -33,11 +33,21 @@
     assert_match( /ensure => 'running'/, result.stdout, "err: #{agent}")
   end
 
+  step "SMF: ensure non-existent services return :absent"
+  on(agent, puppet("resource service bogus")) do
+    assert_match( /ensure => 'absent'/, result.stdout, "err: #{agent}")
+  end
+
   step "SMF: ensure you can stop the service"
   apply_manifest_on(agent, 'service {tstapp : ensure=>stopped}') do
     assert_match( /changed 'running'.* to 'stopped'/, result.stdout, "err: #{agent}")
   end
 
+  step "SMF: ensure stopping a non-existent service doesn't throw an error"
+  apply_manifest_on(agent, 'service {bogus : ensure=>stopped}') do
+    assert_equal(exit_code, 0, "'puppet resource service' should have an exit code of 0")
```

Comment:
I think the `apply_manifest_on` call will fail if the call returns non-zero, so the inner assert will never be reached? Based on https://github.com/puppetlabs/beaker-puppet/blob/3a9eca180daed3d3b3e0d69ea15955039461ac6f/lib/beaker-puppet/helpers/puppet_helpers.rb#L433 it looks like it will default to only accepting 0.

Reply:
@haus good catch. I've pushed an update that includes :acceptable_exit_codes => [0,1,2,4,6]. If I'm understanding things correctly, that should allow the puppet apply command to return all of its possible exit codes. Please correct me if I'm wrong.

Reply:
Cool, that looks good to me.